R/install-remote.R

```diff
@@ -31,12 +28,28 @@ install_remotes <- function(remotes, ...) {
 
 # Add metadata
 add_metadata <- function(pkg_path, meta) {
-  path <- file.path(pkg_path, "DESCRIPTION")
-  desc <- read_dcf(path)
+  source_desc <- file.path(pkg_path, "DESCRIPTION")
+  binary_desc <- file.path(pkg_path, "Meta", "package.rds")
+  if (file.exists(source_desc)) {
+    desc <- read_dcf(source_desc)
+
+    desc <- modifyList(desc, meta)
+
+    write_dcf(source_desc, desc)
+  }
```

Comment:
Unless I am mistaken, that if else structure you suggested will NOT modify BOTH files if they are both present. This modifies either one if they are present, which is what we want to happen, because different functions will read DESCRIPTION, others read the `rsd` file.

You proposed:
```
if_else_function <- function(e1, e2){
  if (e1) {
    print("e1")
  } else if (e2) {
    print("e2")
  } else {
    print("no e1 or e2")
  }
}
```

which gives:

```
> if_else_function(TRUE, FALSE)
[1] "e1"
> if_else_function(TRUE, TRUE)
[1] "e1"
> if_else_function(FALSE, TRUE)
[1] "e2"
```

I (and @jimhester) think we want

```
if_else_function2 <- function(e1, e2){
  if (e1) {
    print("e1")
  } 
  if (e2) {
    print("e2")
  }
  
  if (!e1 && !e2) {
    print("no e1 or e2")
  }
}
```

which gives

```
> if_else_function2(TRUE, FALSE)
[1] "e1"
> if_else_function2(TRUE, TRUE)
[1] "e1"
[1] "e2"
> if_else_function2(FALSE, TRUE)
[1] "e2"
> if_else_function2(FALSE, FALSE)
[1] "no e1 or e2"
```